lm_eval/evaluator.py

```diff
@@ -516,6 +522,21 @@ def evaluate(
                     # or `task_name: []`.
                     # we only want to operate on groups here.
                     continue
+
+                _higher_is_better = deepcopy(higher_is_better[task_list[0]])
+                for task in task_list:
+                    for m, h in higher_is_better[task].items():
+                        if (
+                            m in _higher_is_better
+                            and _higher_is_better[m] is not None
+                            and _higher_is_better[m] != h
+                        ):
```

Comment:
```suggestion
                _higher_is_better = {}
                for task in task_list:
                    for m, h in higher_is_better[task].items():
                    	if m not in _higher_is_better.keys():
                    	    _higher_is_better[m] = h
                        if (
                            m in _higher_is_better
                            and _higher_is_better[m] is not None
                            and _higher_is_better[m] != h
                        ):
```

Reply:
May not be safe to reference `task_list[0]` to obtain all of a group's metrics. 

#1741 should make this cleaner but for now, I believe this should do the trick

Reply:
tested with group `pythia` and this change appears to work!